modules/ravestate_telegramio/telegram_bot.py

```diff
@@ -116,42 +134,125 @@ def handle_photo(bot: Bot, update: Update):
+
+    def add_new_child_process(chat_id):
+        """
+        Adds the chat of the incoming message to the set of active_chats
+        Creates new Ravestate Context in new Process for the new chat and
+        sets up a bidirectional Pipe for communication between Master and Child Processes
+        """
+        # start method has to be 'spawn'
+        mp_context = mp.get_context('spawn')
+        # Pipe to communicate between Master Process and all children
+        parent_conn, child_conn = mp.Pipe()
+        # create commandline args for child config file
+        args = []
+        for child_config_path in child_config_paths_list:
+            args += ['-f', child_config_path]
+        # set up new Process and override child_conn with the Pipe-Connection
+        p = mp_context.Process(target=create_and_run_context,
+                               args=(*args,),
+                               kwargs={'runtime_overrides': [(MODULE_NAME, CHILD_CONN_CONFIG_KEY, child_conn)]})
+        p.start()
+        active_chats[chat_id] = parent_conn
+
     def error(bot: Bot, update: Update, error: TelegramError):
         """
         Log Errors caused by Updates.
         """
         logger.warning('Update "%s" caused error "%s"', update, error)
 
-    """Start the bot."""
-    # Create the EventHandler and pass it your bots token.
-    token = ctx.conf(key=TOKEN_CONFIG_KEY)
-    if not token:
-        logger.error('telegram-token is not set. Shutting down telegramio')
-        return Delete()
-
-    updater: Updater = Updater(token)
-    # Get the dispatcher to register handlers
-    dp: Dispatcher = updater.dispatcher
-    # handle noncommand-messages with handle_input
-    dp.add_handler(MessageHandler(Filters.text, handle_text_input))
-    dp.add_handler(MessageHandler(Filters.photo, handle_photo))
-    # log all errors
-    dp.add_error_handler(error)
-    # Start the Bot
-    updater.start_polling()
+    child_conn = ctx.conf(key=CHILD_CONN_CONFIG_KEY)
+    is_master_process = child_conn is None
+    if is_master_process:
+        # Master Process -> Start the bot
+        # Create the EventHandler and pass it your bots token.
+        token = ctx.conf(key=TOKEN_CONFIG_KEY)
+        if not token:
+            logger.error(f'{TOKEN_CONFIG_KEY} is not set. Shutting down telegramio')
+            return Delete()
+        child_config_paths_list = ctx.conf(key=CHILD_FILES_CONFIG_KEY)
+        if not ctx.conf(key=ALL_IN_ONE_CONTEXT_CONFIG_KEY) and (
+                not child_config_paths_list or not isinstance(child_config_paths_list, list)
+                or not all(os.path.isfile(child_config_path) for child_config_path in child_config_paths_list)):
+            logger.error(f'{CHILD_FILES_CONFIG_KEY} is not set (correctly). Shutting down telegramio')
+            return Delete()
+
+        updater: Updater = Updater(token)
+        # Get the dispatcher to register handlers
+        dp: Dispatcher = updater.dispatcher
```

Comment:
In the world divided by different opinions about naming variables...
I really think naming something dp won't help anybody in understanding your code ;)